modin/config/envvars.py

```diff
@@ -557,12 +549,40 @@ def _get(cls) -> dict:
             Decoded and verified config value.
         """
         custom_parameters = super().get()
-        result = cls.default.copy()
+        result = cls._get_default().copy()
         result.update(
             {key.replace("-", "_"): value for key, value in custom_parameters.items()}
         )
         return result
 
+    @classmethod
+    def _get_default(cls) -> Any:
+        """
+        Get default value of the config. Checks the pyhdk version and omits variables unsupported in prior versions.
+
+        Returns
+        -------
+        dict
+            Config keys and corresponding values.
+        """
+        if (default := getattr(cls, "default", None)) is None:
+            import pyhdk
+            from packaging import version
+
+            cls.default = default = {
+                "enable_union": 1,
+                "enable_columnar_output": 1,
+                "enable_lazy_fetch": 0,
+                "null_div_by_zero": 1,
+                "enable_watchdog": 0,
+                "enable_thrift_logs": 0,
+            }
+
+            if version.parse(pyhdk.__version__) >= version.parse("0.6.1"):
```

Comment:
Let's wait for https://github.com/modin-project/modin/pull/6328 to be merged. After that, we can remove `pyhdk.__version__` and this will fix the test errors.

Reply:
We introduced the pyhdk version API specifically for use in modin to allow us to introduce and use new features without tightly coupling the two releases. If we can guarantee the two versions always match, then perhaps we can remove this version checking. But, I was under the impression that was not something we could guarantee (@ienkovich @Garra1980). 

To fix the test error I can simply import pyhdk under a try/except and ignore the config params if either the pyhdk version is not high enough or if the module import fails.

Reply:
It turns out that depending on the version, and whether it is installed in the environment or not, different default parameters will be used?

@Garra1980 which of the options is supposed to be used to generate documentation?

Reply:
I think for the latest supported pyhdk… in general, @alexbaden is there a place in hdk docs where all options are  described? I think we need to post a link to that place somewhere in modin docs

Reply:
I set the version to 999 as suggested.